core/species.cpp

```diff
@@ -3854,126 +4026,405 @@ void Species::SimplifyTreeSequence(void)
+			tsk_id_t *edge_child = chromosome_tables.edges.child;
+			tsk_id_t *edge_parent = chromosome_tables.edges.parent;
+			tsk_size_t edges_num_rows = chromosome_tables.edges.num_rows;
+			
+			for (tsk_size_t k = 0; k < edges_num_rows; k++) {
+				edge_child[k] = node_id_map[edge_child[k]];
+				edge_parent[k] = node_id_map[edge_parent[k]];
+			}
+			
+#if DEBUG
+			ret = tsk_table_collection_check_integrity(&chromosome_tables, 0);
+			if (ret < 0) handle_error("SimplifyAllTreeSequences() tsk_table_collection_check_integrity after node remapping", ret);
+#endif
+		}
+		
+		// update map of remembered_nodes_; with a single chromosome and a standard simplify,
+		// they would now be the first n entries in the node table (and we used to assume that),
+		// but now that is not guaranteed, and we need to remap them using node_id_map
+		for (tsk_id_t j = 0; j < (tsk_id_t)remembered_nodes_.size(); j++)
+			remembered_nodes_[j] = node_id_map[samples[j]];
+		
+		// and update the tskit node id base for all extant individuals, similarly
+		for (auto subpop_iter : population_.subpops_)
+		{
+			Subpopulation *subpop = subpop_iter.second;
+			
+			for (Individual *ind : subpop->parent_individuals_)
+			{
+				// all the haplosomes for an individual share the same two tskit node ids (shared node table)
+				// we thus need to just remap the base id, and the second id should always remap with it
+				tsk_id_t tsk_node_id_base = ind->TskitNodeIdBase();
+				tsk_id_t remapped_base = node_id_map[tsk_node_id_base];
+				
+				ind->SetTskitNodeIdBase(remapped_base);
+				
+#if DEBUG
+				// check that the second id did remap alongside the first id
+				if (node_id_map[tsk_node_id_base + 1] != remapped_base + 1)
+					EIDOS_TERMINATION << "ERROR (Species::SimplifyAllTreeSequences): node table filtering did not preserve order!" << EidosTerminate(nullptr);
+#endif
+			}
+		}
+		
+		free(keep_nodes);
+		free(node_id_map);
+	}
+	
+	// the individual table needs to be filtered now; we no longer pass TSK_SIMPLIFY_FILTER_INDIVIDUALS for simplification,
+	// so it could be parallelized.  The code here is based on the node table filtering above, mutatis mutandis
+	{
+		int ret;
+		const tsk_size_t num_individuals = main_tables.individuals.num_rows;
+		tsk_bool_t *keep_individuals = (tsk_bool_t *)calloc(num_individuals, sizeof(tsk_bool_t));	// note: cleared by calloc
+		tsk_id_t *individual_id_map = (tsk_id_t *)malloc(num_individuals * sizeof(tsk_id_t));
+		
+		if (!keep_individuals || !individual_id_map)
+			EIDOS_TERMINATION << "ERROR (Species::SimplifyAllTreeSequences): allocation failed; you may need to raise the memory limit for SLiM." << EidosTerminate(nullptr);
+		
+		// mark the individuals we want to keep: all individuals referenced by nodes; note that the node table is shared,
+		// so we only need to loop through that one shared node table that is kept by the main table collection
+		{
+			tsk_id_t *node_individual = main_tables.nodes.individual;
```

Comment:
minor style nit: elsewhere (eg in tskit) we'd call this `nodes_individual` so it's parallel to `nodel.individual` rather than `node_individual` (or `node_individuals`); nbd obvs tho

Reply:
Sure.  Please point this out anywhere you notice it, and I'll try to get into the habit.

Reply:
I notice that tskit is actually not consistent about this.  I searched on `node_individual` to find places to clean up, and it found two places where tskit uses that instead of `nodes_individual` (at least in SLiM's outdated copy of the tskit code).  Just FYI.  I will fix it in SLiM nevertheless.  :->

Reply:
Done, resolving.